src/ConnectedMode/Migration/IMigrationPrompt.cs

```diff
@@ -79,14 +88,16 @@ public async Task ShowAsync()
             });
         }
 
+        private void OnMigrationWIzardFinished(object sender, EventArgs e) => Clear();
+        
         public void Clear()
         {
             notificationService.RemoveNotification();
         }
 
         private void OnMigrate()
         {
-            // TODO: Show migration wizard
+            migrationWizardController.StartMigrationWizard();
```

Comment:
With this implementation, the gold bar will remain visible until the migration wizard has finished.

Would it make much difference from a UX point of view if we removed it immediately i.e. as soon as the wizard has started? That would make the code simpler because we wouldn't need to listen for the "finished" event.

Reply:
But what if the user closes the migration window before actively starting the migration process?

Reply:
Ok: so the UX would be:

* solution opens, migration is required -> gold bar shown.

**Scenario 1 :** user closes the gold bar without clicking on any buttons.
The migration check will happen again next time the user opens the solution.

**Scenario 2:** users clicks on `Migrate` -> wizard shown. User clicks `Cancel`.
Gold bar is still visible. User can trigger the wizard again or close the gold bar.

**Scenario 3:** users clicks on `Migrate` -> wizard shown. User clicks `Start` but migration is not successful.
Same as (2) i.e. gold bar is still visible. User can trigger the wizard again or close the gold bar.

**Scenario 4:** users clicks on `Migrate` -> wizard shown. User clicks `Start`, migration is successful.
Gold bar is closed when the wizard closes.

Is that right? If so, seems ok to me.

Reply:
Yes, that is correct. The only problem would be Scenario 3 (depending on how far the migration got) but I think we need to deal with that seperately anyways.